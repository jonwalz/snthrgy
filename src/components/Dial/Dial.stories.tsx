import * as React from "react";
import {MainDial} from "./Dial";

export default {
  title: "Dial",
  component: MainDial,
};

export const Dial = () => <MainDial />;
