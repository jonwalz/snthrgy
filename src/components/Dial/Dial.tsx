import React from "react";
import Knob, {KnobProps, composeTwo, useAngleUpdater} from "react-dial-knob";
import {ReactComponent as DialSvg} from "../../assets/dial.svg";
import {Container} from "./styles";

export interface DialProps extends KnobProps {
  style?: React.CSSProperties;
}

export const Dial = (props: DialProps): JSX.Element => {
  const [angle, setAngle] = useAngleUpdater(props.value);
  const angleChangeHandler = composeTwo<number>(setAngle, props.onAngleChange);

  return (
    <Container>
      <Knob
        diameter={props.diameter}
        value={props.value}
        min={props.min}
        max={props.max}
        step={props.step}
        spaceMaxFromZero={props.spaceMaxFromZero}
        ariaLabelledBy={props.ariaLabelledBy}
        ariaValueText={props.ariaValueText}
        knobStyle={{cursor: "pointer", ...props.knobStyle}}
        onAngleChange={angleChangeHandler}
        onInteractionChange={props.onInteractionChange}
        onValueChange={props.onValueChange}
      >
        <DialSvg
          transform={`rotate(${angle})`}
          style={{transform: `rotate(${angle}deg)`}}
        />
      </Knob>
      {props.children}
    </Container>
  );
};
