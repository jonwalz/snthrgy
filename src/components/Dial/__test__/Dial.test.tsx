import * as React from "react";
import {render} from "@testing-library/react";
import {Dial} from "../Dial";

const props = {
  diameter: 10,
  min: 0,
  max: 10,
  step: 1,
  value: 5,
  onValueChange: () => null,
};

describe("Dial", () => {
  it("should match snapshot", () => {
    const wrapper = render(<Dial {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
