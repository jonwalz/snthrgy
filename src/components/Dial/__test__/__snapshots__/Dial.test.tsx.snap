// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dial should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ jsEWwN"
      >
        <div
          aria-valuemax="10"
          aria-valuemin="0"
          aria-valuenow="5"
          style="width: 10px; height: 10px; border-radius: 5px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
          tabindex="0"
        >
          <svg
            style="transform: rotate(164deg);"
            transform="rotate(164)"
          >
            dial.svg
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ jsEWwN"
    >
      <div
        aria-valuemax="10"
        aria-valuemin="0"
        aria-valuenow="5"
        style="width: 10px; height: 10px; border-radius: 5px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
        tabindex="0"
      >
        <svg
          style="transform: rotate(164deg);"
          transform="rotate(164)"
        >
          dial.svg
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
